src/util/Optional.h

```diff
@@ -42,13 +42,17 @@ class Optional {
 
     T& value()
     {
-        ASSERT(m_hasValue);
+#if defined(NDEBUG)
+        assert(m_hasValue);
+#endif
         return m_value;
     }
 
     const T& value() const
     {
-        ASSERT(m_hasValue);
+#if defined(NDEBUG)
```

Comment:
IMO you can remove #if defined(NDEBUG)
since C standard ISO defines behavior with NDEBUG and assert
```
The header <assert.h> defines the assert macro and refers to another macro,
NDEBUG
which is not defined by <assert.h>. If NDEBUG is defined as a macro name at the
point in the source file where <assert.h> is included, the assert macro is defined
simply as
#define assert(ignore) ((void)0)
```

see 7.2 of ISO 9899:1999

Reply:
Thanks for the review!
All `NDEBUG` checks are removed :)